specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json

```diff
@@ -2189,7 +2189,7 @@
       "type": "string",
       "in": "header",
       "required": true,
-      "x-ms-parameter-location": "client",
+      "x-ms-parameter-location": "method",
```

Comment:
I think SDK don't want this parameter at all. All SDK has built-in policy on `user-agent`.

However, I won't be able to merge this.

Reply:
Agreed with weidong, for just we would always add the user-agent header by default. If you want to custom it we have a client-level option to set prefix.

```
{
userAgentOptions: { userAgentPrefix: "AdvancedSample V1.0.0" }, // Customized user-agent string
}

```

Reply:
All SDK has user-agent built-in. It takes below form by default:
https://azure.github.io/azure-sdk/general_azurecore.html#telemetry-policy

PS: even `curl` sets the user-agent. So, just don't put this to swagger...

Reply:
I may not get what you mean by "suffix route wouldn't work". Synapse **requires** a certain user-agent from user to use the API?

Reply:
What I meant by suffix route not working is that we will to make a code change in the service to process the header with a suffix. However, we wouldn't need this as I have removed the user-agent param completely from the swagger. See below


Removed the user-agent param entirely from the Swagger - see this [PR](https://github.com/Azure/azure-rest-api-specs/pull/24469)

Resolving this conversation.